NeoML/src/TraditionalML/SparseFloatMatrix.cpp

```diff
@@ -76,30 +75,36 @@ CSparseFloatMatrix::CSparseFloatMatrixBody::CSparseFloatMatrixBody( const CFloat
 				}
 			}
 		}
-		ColumnsBuf.SetBufferSize( ElementsBufferSize );
-		ValuesBuf.SetBufferSize( ElementsBufferSize );
+		Desc.Columns = new int[ElementsBufferSize];
+		Desc.Values = new float[ElementsBufferSize];
 		for( int i = 0; i < desc.Height; ++i ) {
 			Desc.PointerB[i] = ElementCount;
 			for( int pos = desc.PointerB[i], j = 0; pos < desc.PointerE[i]; ++pos, ++j ) {
 				if( desc.Values[pos] != 0 ) {
-					ColumnsBuf.Add( j );
-					ValuesBuf.Add( desc.Values[pos] );
+					Desc.Columns[ElementCount] = j;
+					Desc.Values[ElementCount] = desc.Values[pos];
 					++ElementCount;
 				}
 			}
 			Desc.PointerE[i] = ElementCount;
 		}
 	} else {
-		ColumnsBuf.SetSize( ElementsBufferSize );
-		ValuesBuf.SetSize( ElementsBufferSize );
-		::memcpy( ColumnsBuf.GetPtr(), desc.Columns, ElementsBufferSize * sizeof( int ) );
-		::memcpy( ValuesBuf.GetPtr(), desc.Values, ElementsBufferSize * sizeof( float ) );
-		::memcpy( BeginPointersBuf.GetPtr(), desc.PointerB, RowsBufferSize * sizeof( int ) );
-		::memcpy( EndPointersBuf.GetPtr(), desc.PointerE, RowsBufferSize * sizeof( int ) );
+		Desc.Columns = new int[ElementsBufferSize];
+		Desc.Values = new float[ElementsBufferSize];
+		CopyDescData( Desc, desc, ElementCount );
```

Comment:
You should copy by vector! Because if you have a submatrix, ElementCount will be wrong!

Reply:
I guess ElementCount will be OK in case of submatrix too

Reply:
caught it, fixed.